my_types.py

```diff
@@ -0,0 +1,9 @@
+"""
+A bunch of type definitions to improve readability
```

Comment:
might be simpler to just have that in shared.py?

Reply:
I attempted that, but there is a cyclic dependency between `shared.py` and `setup.py` which makes things messy.

See 8286a2f for an attempt (but I had to duplicate the types in `setup.py`). I can revert if you don't like.

Reply:
Ah, I see the problem now. On second thought it does indeed seem cleaner to keep the typing separate in its own file then. Would you mind reverting that last commit?